include/any_wrappers.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+#include <vector>
+#include <any>
+#include "tsl/robin_set.h"
+
+namespace AnyWrapper
+{
+struct AnyRobinSet
+{
+    template <typename T>
+    AnyRobinSet(const tsl::robin_set<T> &robin_set) : _data(const_cast<tsl::robin_set<T> *>(&robin_set))
```

Comment:
I tied this approach, but problem being AnyRobinSet<> would still need the data_type T, and our functions are pure virtual (non-templated ofcourse). How do we get that Type Info ?

Reply:
Ah, right, sorry, I forgot that we only want the type in the constructor so this won't work.

My main concern is `const_cast` because you can construct an object using a const `AnyVector` and then use a `get` method which gives you a non-const object. That should not happen.

So perhaps using `AnyContainerRef` (or even making it more generic, into a `AnyReference`), inheriting `AnyVector` from `AnyReference` and making `AnyVector` just have a constructor which calls the parent `AnyReference` constructor would work for this purpose instead.